htdocs/compta/facture/class/facture-rec.class.php

```diff
@@ -1032,42 +1032,45 @@ function createRecurringInvoices($restictoninvoiceid=0, $forcevalidation=0)
 
 					dol_syslog("createRecurringInvoices Process invoice template id=".$facturerec->id.", ref=".$facturerec->ref.", entity=".$facturerec->entity);
 
-				    $facture = new Facture($db);
+					$facture = new Facture($db);
 					$facture->fac_rec = $facturerec->id;    // We will create $facture from this recurring invoice
 					$facture->fk_fac_rec_source = $facturerec->id;    // We will create $facture from this recurring invoice
 
-				    $facture->type = self::TYPE_STANDARD;
-				    $facture->brouillon = 1;
-				    $facture->date = (empty($facturerec->date_when)?$now:$facturerec->date_when);	// We could also use dol_now here but we prefer date_when so invoice has real date when we would like even if we generate later.
-				    $facture->socid = $facturerec->socid;
-
-				    $invoiceidgenerated = $facture->create($user);
-				    if ($invoiceidgenerated <= 0)
-				    {
-				        $this->errors = $facture->errors;
-				        $this->error = $facture->error;
-				        $error++;
-				    }
-				    if (! $error && ($facturerec->auto_validate || $forcevalidation))
-				    {
-				        $result = $facture->validate($user);
-				        if ($result <= 0)
-				        {
-	    			        $this->errors = $facture->errors;
-	    			        $this->error = $facture->error;
-				            $error++;
-	                    }
-				    }
-	                if (! $error && $facturerec->generate_pdf)
-	                {
-	                    $result = $facture->generateDocument($facturerec->modelpdf, $langs);
-	                    if ($result <= 0)
-	                    {
-	                        $this->errors = $facture->errors;
-	                        $this->error = $facture->error;
-	                        $error++;
-	                    }
-	                }
+					$facture->type = self::TYPE_STANDARD;
+					$facture->brouillon = 1;
+					$facture->date = (empty($facturerec->date_when)?$now:$facturerec->date_when);	// We could also use dol_now here but we prefer date_when so invoice has real date when we would like even if we generate later.
+					$facture->socid = $facturerec->socid;
+
+					$invoiceidgenerated = $facture->create($user);
+					if ($invoiceidgenerated <= 0)
+					{
+						$this->errors = $facture->errors;
+						$this->error = $facture->error;
+						$error++;
+					}
+					if (! $error && ($facturerec->auto_validate || $forcevalidation))
+					{
+						$result = $facture->validate($user);
+						if ($result <= 0)
+						{
+							$this->errors = $facture->errors;
+							$this->error = $facture->error;
+							$error++;
+						}
+					}
+					if (! $error && $facturerec->generate_pdf)
+					{
+						$result = $facture->generateDocument($facturerec->modelpdf, $langs);
+						if ($result <= 0)
+						{
+							$this->errors = $facture->errors;
+							$this->error = $facture->error;
+							$error++;
+						}
+					}
+					if (! $error) {
+						$facture->call_trigger('BILL_VALIDATE', $user);
```

Comment:
The trigger to validate an invoice is already called into the validate method of invoice.

Reply:
Oh yes, that right, I need to do a bigger PR, I add to move BILL_VALIDATE from invoice validate method, because pdf generation was trigger after that process, and I send invoice in pdf directly by email with notification.
I also add a BILL_REMINDER, I can add too if you want.

About notify class, I don't know why does this line exists : 
https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/core/class/notify.class.php#L480

$filename isn't an array (I rewrite a part of this too)